wolfcrypt/src/aes.c

```diff
@@ -3702,6 +3724,7 @@ int wc_AesSetIV(Aes* aes, const byte* iv)
             #endif /* WOLFSSL_AESNI_BYx */
             /* store iv for next call */
             XMEMCPY(aes->reg, aes->tmp, AES_BLOCK_SIZE);
+            RESTORE_VECTOR_REGISTERS();
```

Comment:
Indeed as currently compiled (`-fno-builtin`), the `memcpy()`s don't need to be protected.  I've tweaked it at the codepoint you identify, and at several others, to reflect this consistently.

With `-fbuiltin`, `memcpy()` gets vectorized, and in the early going (reverse-assembly-guided) they wound up protected on that basis.  But there are many other unprotected `memcpy()`s there, so there's no point protecting just these few of them that happen to be adjacent to SAVE_/RESTORE_ for other (AESNI and vectorized asm) reasons.

Note that while protected, the kthread is non-preemptable, so that's another excellent reason to minimize the protected spans.

Reply:
Oh and all the inner #if cases are vectorized so it seemed neater to SAVE_/RESTORE_ right outside the #if...#endif.